python/tvm/micro/compiler.py

```diff
@@ -1,361 +0,0 @@
-# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
maybe keep this and add warning that it will be removed in the next version?

Reply:
so i was thinking about this, one thing is that i had to make breaking changes to the `apps/microtvm/zephyr` code. i don't mind leaving in the old `tvm.micro.Compiler` stuff (though I'd need to adapt more Transport layers), but it might be confusing to have two different styles of Zephyr template projects. we haven't done a TVM release with this code in there, so i wonder if it's necessary to keep the Compiler stuff in there. wdyt?

Reply:
@mehrdadh any thoughts here?

Reply:
@areusch  I agree that since there's no official release we don't commit to support older stuff. However I recommend to make it more clear in the PR title/description that this PR is removing certain modules so if anyone has a problem could raise their voice.
The other thing is that you could announce it in the microtvm community meeting and get some feedback there. My concern is that people have internal projects depending on these features.